src/dreamer/dream_scheduler.py

```
@@ -202,12 +202,33 @@ async def execute_dream(
+                    models.Document.workspace_name == workspace_name,
+                    models.Document.observer == observer,
+                    models.Document.observed == observed,
+                )
+                .order_by(models.Document.created_at.desc())
+                .limit(1)
+            )
+            session_name = await db.scalar(stmt)
+
+        if not session_name:
+            logger.warning(
+                f"No documents found for {workspace_name}/{observer}/{observed}, skipping dream"
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)